hw3/02-charts.js

```diff
@@ -1,52 +1,296 @@
+	"rgba(78, 52, 199, 0.8)", //tyrell
+	"rgba(54, 54, 54, 0.8)", // free folk
+	"rgba(255, 206, 186, 0.8)", //tarth
+	"rgba(255, 99, 132, 0.8)", // naathi
+	"rgba(75, 275, 275, 0.8)", // bolton
+	"rgba(100, 70, 70, 0.8)", // naharis
+	"rgba(255, 159, 64, 0.8)", // lorath
+	"rgba(199, 100, 199, 0.8)", // Mormont
+	"rgba(183, 255, 255, 0.8)", // sparrow
+	"rgba(40, 259, 164, 0.8)", // viper
+	"rgba(210, 200, 100, 0.8)", //sand
+	"rgba(128, 0, 128, 0.8)", // worm
+	"rgba(255, 15, 150, 0.8)", // qyburn
+	"rgba(0, 128, 0, 0.8)", // bronn
 ];
 
 const borderColors = [
-  'rgba(54, 162, 235, 1)',
-  'rgba(255, 206, 86, 1)',
-  'rgba(255, 99, 132, 1)',
-  'rgba(75, 192, 192, 1)',
-  'rgba(153, 102, 255, 1)',
-  'rgba(255, 159, 64, 1)',
-  'rgba(159, 159, 159, 1)',
-  'rgba(83, 102, 255, 1)',
-  'rgba(40, 159, 64, 1)',
-  'rgba(210, 199, 199, 1)',
-  'rgba(78, 52, 199, 1)',
+	"rgba(54, 162, 235, 0.8)", //targaryen
+	"rgba(255, 206, 86, 0.8)", //tarly
+	"rgba(255, 99, 132, 0.8)", //stark
+	"rgba(75, 192, 192, 0.8)", //baratheon
+	"rgba(153, 102, 255, 0.8)", // lannister
+	"rgba(255, 159, 64, 0.8)", // grejoy
+	"rgba(199, 199, 199, 0.8)", // clegane
+	"rgba(83, 102, 255, 0.8)", //house belish
+	"rgba(40, 159, 64, 0.8)", //seaworth
+	"rgba(150, 150, 199, 0.8)", /// unknown
+	"rgba(78, 52, 199, 0.8)", //tyrell
+	"rgba(54, 54, 54, 0.8)", // free folk
+	"rgba(255, 206, 186, 0.8)", //tarth
+	"rgba(255, 99, 132, 0.8)", // naathi
+	"rgba(75, 275, 275, 0.8)", // bolton
+	"rgba(100, 70, 70, 0.8)", // naharis
+	"rgba(255, 159, 64, 0.8)", // lorath
+	"rgba(199, 100, 199, 0.8)", // Mormont
+	"rgba(183, 255, 255, 0.8)", // sparrow
+	"rgba(40, 259, 164, 0.8)", // viper
+	"rgba(210, 200, 100, 0.8)", //sand
+	"rgba(128, 0, 128, 0.8)", // worm
+	"rgba(255, 15, 150, 0.8)", // qyburn
+	"rgba(0, 128, 0, 0.8)", // bronn
 ];
 
+//Algorithm:
+//*fetched the data
+//*mapped the family names of all the characters.
+//*used hashmap to find unique family names and
+//group, the desired family names together.
+//*stored the keys and values obtained from hashmap into
+//labels and values arrays.
+//*Passed these arrays to the function generating donut chart
+//*used these arrays to get the labels and data for donut chart
+
 // url for the Thrones API
-const url = 'https://thronesapi.com/api/v2/Characters';
-
-const renderChart = () => {
-  const donutChart = document.querySelector('.donut-chart');
-
-  new Chart(donutChart, {
-    type: 'doughnut',
-    data: {
-      labels: ['label', 'label', 'label', 'label'],
-      datasets: [
-        {
-          label: 'My First Dataset',
-          data: [1, 12, 33, 5],
-          backgroundColor: backgroundColors,
-          borderColor: borderColors,
-          borderWidth: 1,
-        },
-      ],
-    },
-  });
+const url = "https://thronesapi.com/api/v2/Characters";
+
+const getData = async () => {
+	await fetch(url)
+		.then((response) => {
+			return response.json();
+		})
+		.then((data) => {
+			mapToHouses(data);
+			
+		});
```

Comment:
I think this could use a catch

Reply:
Ya, should have used catch. I will try adding it and push the code again.